ngx_http_sxg_filter_module.c

```diff
@@ -819,20 +819,39 @@ static ngx_int_t ngx_http_sxg_body_filter_impl(ngx_http_request_t* req,
     return NGX_AGAIN;
   }
 
-  // Copy SXG headers to innerHTML header
-  if (!copy_response_header_to_sxg_header(req->pool, &req->headers_out.headers,
+  // Copy SXG headers to innerHTML header.
+  if (!sxg_header_append_string("content-type",
+                                (const char*)req->headers_out.content_type.data,
+                                &ctx->response.header) ||
+      !copy_response_header_to_sxg_header(req->pool, &req->headers_out.headers,
                                           &ctx->response.header)) {
+    ngx_log_error(NGX_LOG_ERR, req->connection->log, 0,
+                  "nginx-sxg-module: failed to copy headers to SXG");
     sxg_raw_response_release(&ctx->response);
     return NGX_ERROR;
   }
 
-  sxg_header_append_string("content-type",
-                           (const char*)req->headers_out.content_type.data,
-                           &ctx->response.header);
-  static const char kSxgContentType[] = "application/signed-exchange;v=b3";
-  ngx_str_set(&req->headers_out.content_type, kSxgContentType);  // must be SXG
+  ngx_table_elt_t* server = req->headers_out.server;
+  // Cleanup outer header.
+  ngx_memzero(&req->headers_out, sizeof(ngx_http_headers_out_t));
+  if (ngx_list_init(&req->headers_out.headers, req->pool, 1,
```

Comment:
Do we also need to do this for headers_out.trailers?

Reply:
Done.